dynamiqs/solvers/ode/adaptive_solver.py

```diff
@@ -201,18 +201,13 @@ def integrate_augmented(
 
             # the computation graph attached to `y` and `a` is automatically freed after
             # next line, because there are no more references to the original tensors
-            # (the old `y`, `a`, `fyt` and `fat` go out of scope)
-            y, a, fyt, fat = (
-                new_leaf_tensor(y),
-                new_leaf_tensor(a),
-                new_leaf_tensor(fyt),
-                new_leaf_tensor(fat),
-            )
+            # (the old `y` and `a` go out of scope)
+            y, a = new_leaf_tensor(y), new_leaf_tensor(a)
```

Comment:
Aren't we going to store a large computation graph to track `fyt` and `fat`?

Reply:
TBH I don't understand this part so well. For sure if you add back the `new_leaf_tensor` call on `fyt` and `fat` the tests don't pass anymore. Also, it doesn't seem like the run time of the backward pass explodes, so I think we're not passing through the graph of `fyt` and `fat` multiple times. 

I'm guessing they go out of scope naturally and so their graph is detached by PyTorch.

Reply:
Right, but I think it would be great if we could make it explicit using the `new_leaf_tensor` function, or a comment when it goes out of scope. Otherwise it's very empirical, and unreadable for someone who don't know the code. We can do this in another PR though, nothing urgent.